### ISSUE TITLE
Setting footer.display to false not working

### ISSUE THREAD

**Issue by e0lithic** [Mister-Hope/vuepress-theme-hope#170]
Hi, 

- The documentation says that the footer.display config has a default value of false, however still the footer shows on all pages
- Setting footer.display to false in config.js has no effects and still the footer is displayed.

Thanks

**Comment by Mister-Hope** [comment]
I will have a check later.

**Comment by e0lithic** [comment]
@Mister-Hope In the new release, footer.display is set to true by default. If in such a case, if we use `footer: { display: false}` in config.js and then add `footer: true` to enable it locally in the `frontmatter` of the home page. It doesn't display the footer at all. Usecase is to add the footnote to only the home page. Can you confirm please.

**Comment by Mister-Hope** [comment]
Nope, perhaps I have some mistakes in the docs. The footer key in page corresponds to `themeConfig.footer.content`. I don't think users will like to write like this:

```
---
footer:
  copyright: xxx
  content: xxx
  mediaLink: xxx
---
```

So I use footer instead of content in the frontmatter key.

**Comment by Mister-Hope** [comment]
Maybe I should change it to:

```
---
footerDisplay: xxx
copyright: xxx
footerContent: xxx
mediaLink: xxx
---
```

I will think about it later.

**Comment by e0lithic** [comment]
I tried the following . However nothing works when `footer: {display: false}` is set in config.js
```
---
home: true
footer: Copyright © 
copyright: Copyright © 
---
```

**Comment by Mister-Hope** [comment]
I will have a check, I just notice I only edit the config part, with the guide part docs wrong.

**Comment by Mister-Hope** [comment]
I found a bug, I will fix it in 0.2.1.

**Comment by e0lithic** [comment]
Meanwhile I will also put a list of potential enhancements for discussion.

**Comment by Mister-Hope** [comment]
Besides, I will spend some time fixing the docs and think of whether changing the configuration.

**Comment by Mister-Hope** [comment]
> Meanwhile I will also put a list of potential enhancements for discussion.


Welcome! :smile:

**Comment by Mister-Hope** [comment]
I just release `0.3.0-alpha.1` on alpha channel. It should fix this issue.

Meanwhile, `0.3.0-alpha.1` is introducing four plugins. Related docs [here](https://vuepress-theme.mrhope.site/en/config/plugin/)

**Comment by e0lithic** [comment]
I will test it soon and let you know.

**Comment by e0lithic** [comment]
Works great.

**Comment by e0lithic** [comment]
It works well in dev mode but when trying to build , I am getting the following error

```
(undefined) /media/e0lithic/D/GameX/src_generic/release_tools/Test2/node_modules/vuepress-theme-hope/src/components/NavLink.vue.ts
[tsl] ERROR in /media/e0lithic/D/GameX/src_generic/release_tools/Test2/node_modules/vuepress-theme-hope/src/components/NavLink.vue.ts(37,32)
      TS2339: Property 'link' does not exist on type 'NavBarConfigItem'.
```

Am I missing something?

**Comment by Mister-Hope** [comment]
> It works well in dev mode but when trying to build , I am getting the following error
> 
> ```
> (undefined) /media/e0lithic/D/GameX/src_generic/release_tools/Test2/node_modules/vuepress-theme-hope/src/components/NavLink.vue.ts
> [tsl] ERROR in /media/e0lithic/D/GameX/src_generic/release_tools/Test2/node_modules/vuepress-theme-hope/src/components/NavLink.vue.ts(37,32)
>       TS2339: Property 'link' does not exist on type 'NavBarConfigItem'.
> ```
> 
> Am I missing something?

Perhaps, I am missing something, my blog fails too on 0.3.0-alpha2 because of the types problem, I will make a full test before publishing 0.3.0

I am facing an issue with types, sometime the theme is working fine on this project, the docs can be built successfully, while the build will fail when it's in the node_modules

**Comment by Mister-Hope** [comment]
Oh, you are not configing both `link` and `items` on the same object, do you? The theme currently does not support that.

**Comment by Mister-Hope** [comment]
There are some bugs which cause the typings of `@mr-hope/shared-utils` failing, I am fixing it now.

**Comment by e0lithic** [comment]
This seems to fixed for my testcases with release 0.3.0-alpha.4

**Comment by e0lithic** [comment]
Although the sitemap is generated. There are some issues in it. Eg. [Sitemap for vuepress-theme-hope](https://vuepress-theme.mrhope.site/sitemap.xml)

**Comment by Mister-Hope** [comment]
> Although the sitemap is generated. There are some issues in it. Eg. [Sitemap for vuepress-theme-hope](https://vuepress-theme.mrhope.site/sitemap.xml)

I will check it out, sitemap is still in dev

**Comment by Mister-Hope** [comment]
> Although the sitemap is generated. There are some issues in it. Eg. [Sitemap for vuepress-theme-hope](https://vuepress-theme.mrhope.site/sitemap.xml)

Thanks again, fixed in `0.3.0-alpha.5`

**Comment by e0lithic** [comment]
Working well
